apps/discord-bot/src/triggers/cta-post.ts

```diff
@@ -0,0 +1,280 @@
+  requireGuild: boolean;
+  public triggered(msg: Message): boolean {
+    // eslint-disable-next-line @typescript-eslint/no-shadow
+    const ctaChannel = msg.guild?.channels.cache.find(ctaChannel => ctaChannel?.name === channelName)
+    // check cta channel exists
+    if (ctaChannel !== undefined) {
+
+      // check message came from cta channel
+      if (msg.channelId === ctaChannel.id) {
+        return true
+      }
+    }
+
+    return false
+  }
+
+  public async execute(msg: Message): Promise<void> {
+
+    if (msg === undefined) {
+      return
+    }
+
+    if (msg.channel.type === ChannelType.GuildText) {
+
+      // check if message created in last month
+      if (msg.createdTimestamp >= d.getMonth() - 1) {
+
+        // check if message already has a thread
+        if (msg.hasThread) {
+
+          if (msg.thread?.archived) {
+            msg.thread.setArchived(false, "Updating chart...")
+          }
+          // is it active?
+          const thread = await this.activeCTAThread(msg)
+          if (thread != undefined) {
+
+            // update the chart in the thread
+            this.updateChart(msg, thread)
+          }
+
+          return
+        }
+
+        // create thread and start collector
+        const thread = await this.createCTAThread(msg)
+        if (thread) {
+          this.startCTAReactionCollector(msg, thread)
+        }
+      }
+    }
+  }
+
+  private async getFinishedReactions(msg: Message): Promise<object> {
+    const finishedReactions = msg.reactions.cache.filter(reaction => finishedEmoji === reaction.emoji.name);
+    const roleReactions: { [region: string]: string[] } = {};
+
+    finishedReactions.forEach(async (reaction) => {
+      const users = await reaction.users.fetch();
+      users.forEach(async (user) => {
+        if (user.id === msg.client.user.id) {
+          return;
+        }
+        const member = msg.guild?.members.fetch(user.id);
+        const memberRoles = (await member)?.roles.cache;
+        const role = memberRoles?.filter(r => regionRoles.includes(r.name));
+        const memberRegionRole = role?.first()?.name;
+
+        if (memberRegionRole != undefined) {
+          if (!roleReactions[memberRegionRole]) {
+            roleReactions[memberRegionRole] = [];
+          }
+          if (!roleReactions[memberRegionRole]?.includes(user.id)) {
+            roleReactions[memberRegionRole]?.push(user.id);
+            console.log(`Adding ${role?.first()?.name} member, ${user.displayName}`);
+          }
+        }
+      });
+    });
+    return roleReactions;
+  }
+
+  private async activeCTAThread(msg: Message): Promise<PublicThreadChannel | undefined> {
+    if (msg.hasThread && msg.thread?.type === ChannelType.PublicThread && !msg.thread?.archived) {
+      return msg.thread;
+    }
+  }
+
+  private async createCTAThread(msg: Message): Promise<PublicThreadChannel | undefined> {
+
+    if (msg.channel.type === ChannelType.GuildText) {
```

Comment:
This is more of a personal preference, but `msg` should always be `ChannelType.GuildText` since the calling function already does that check. Is there a TypeScript error if you don't do this check? If so, I personally prefer either throwing an exception on the invalid condition or making a type assertion. Returning `undefined` in conditions like this can make it hard to determine where logic is falling apart.